src/components/datagrid/data_grid.tsx

```diff
@@ -218,10 +218,16 @@ function renderPagination(props: EuiDataGridProps, controls: string) {
   } = pagination;
   const pageCount = Math.ceil(props.rowCount / pageSize);
 
-  if (props.rowCount < pageSizeOptions[0]) {
+  if (
+    props.rowCount <
+    ((pageSizeOptions && pageSizeOptions.sort((a, b) => a - b)[0]) || pageSize)
```

Comment:
`pageSizeOptions` must hold numbers, no need for a custom comparator to be provided,

```suggestion
    ((pageSizeOptions && pageSizeOptions.sort()[0]) || pageSize)
```

Reply:
The default `Array.prototype.sort` comparator converts array values into strings -> [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Description)

So, `[100, 50, 20, 10, 5]` will be sorted to `[10, 100, 20, 5, 50]` !
That's why the comparator fn is necessary here.

The only thing I lost there, `Array.prototype.sort` mutates an original array, so I created a copy of the `pageSizeOptions` prop and also added comments in the last commit.

Reply:
Ugh you're right, good catch on both of those. 🤦